src/Runner/LintingResultAwareFileIteratorInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner;
+
+use PhpCsFixer\Linter\LintingResultInterface;
+
+/**
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ *
+ * @extends \Iterator<mixed, \SplFileInfo>
+ */
+interface LintingResultAwareFileIteratorInterface extends \Iterator
```

Comment:
mind to describe usecase?
is there a case like we use FileCachingLintingFileIterator but without `lintingResultAware` aspect?
(i wonder if we need this interface)

Reply:
If I remember correctly it's mainly about having common contract that is used in `Runner::getFileIterator()` both in `fixSequential()` and `fixParallel()`, to not duplicate the logic that was previously in `fix()`. Most probably it was also to satisfy PHPStan.

Reply:
is there a case like we use FileCachingLintingFileIterator but without lintingResultAware aspect?

out of the blue it looks to me (but pls fingerprint) that we don't need this interface and we can always check against previously existing `FileLintingFileIterator` and not the new, additional iterator

Reply:
Since `Runner::getFileIterator()` (code moved from `Runner::fix()`) contains this:

```php
return $this->linter->isAsync()
            ? new FileCachingLintingFileIterator($fileFilterIterator, $this->linter)
            : new FileLintingFileIterator($fileFilterIterator, $this->linter);
```

I needed an interface that could be used as a return type. Yes, I could add `@return FileLintingIterator|FileCachingLintingIterator` but thought that native type pointing to common interface would be better. If you don't like it, I can remove the interface and fallback to phpDoc-based union type.

Reply:
ok